src/ucs/config/parser.c

```diff
@@ -2138,68 +2159,62 @@ ucs_config_parser_print_opts_recurs(FILE *stream, const void *opts,
-                                              head->prefix,
-                                              aliased_field->name);
+                ucs_config_parser_print_field(
+                        stream, UCS_PTR_BYTE_OFFSET(opts, alias_table_offset),
+                        prefix, prefix_list, field->name, aliased_field,
+                        flags_p, title, filter, "%-*s %s%s%s",
+                        UCS_CONFIG_PARSER_DOCSTR_WIDTH, "alias of:", prefix,
+                        head->prefix, aliased_field->name);
             }
         } else {
             if (ucs_config_is_deprecated_field(field) &&
-                !(flags & UCS_CONFIG_PRINT_HIDDEN)) {
+                !(*flags_p & UCS_CONFIG_PRINT_HIDDEN)) {
                 continue;
             }
             ucs_config_parser_print_field(stream, opts, prefix, prefix_list,
-                                          field->name, field, flags, NULL);
+                                          field->name, field, flags_p, title,
+                                          filter, NULL);
         }
     }
 }
 
-void ucs_config_parser_print_opts(FILE *stream, const char *title, const void *opts,
-                                  ucs_config_field_t *fields, const char *table_prefix,
-                                  const char *prefix, ucs_config_print_flags_t flags)
+void ucs_config_parser_print_opts(FILE *stream, const char *title,
+                                  const void *opts, ucs_config_field_t *fields,
+                                  const char *table_prefix, const char *prefix,
+                                  const ucs_config_print_flags_t flags,
+                                  const char *filter)
 {
+    ucs_config_print_flags_t flags_copy = flags;
     ucs_config_parser_prefix_t table_prefix_elem;
     UCS_LIST_HEAD(prefix_list);
 
-    if (flags & UCS_CONFIG_PRINT_DOC) {
-        fprintf(stream, "# UCX library configuration file\n");
-        fprintf(stream, "# Uncomment to modify values\n");
-    }
-
-    if (flags & UCS_CONFIG_PRINT_HEADER) {
-        fprintf(stream, "\n");
-        fprintf(stream, "#\n");
-        fprintf(stream, "# %s\n", title);
-        fprintf(stream, "#\n");
-        fprintf(stream, "\n");
-    }
-
     if (flags & UCS_CONFIG_PRINT_CONFIG) {
         table_prefix_elem.prefix = table_prefix ? table_prefix : "";
         ucs_list_add_tail(&prefix_list, &table_prefix_elem.list);
-        ucs_config_parser_print_opts_recurs(stream, opts, fields, flags,
-                                            prefix, &prefix_list);
-    }
-
-    if (flags & UCS_CONFIG_PRINT_HEADER) {
-        fprintf(stream, "\n");
+        ucs_config_parser_print_opts_recurs(stream, opts, fields, &flags_copy,
+                                            prefix, &prefix_list, title,
+                                            filter);
+    } else if (flags & UCS_CONFIG_PRINT_HEADER) {
+        ucs_config_parser_print_header(stream, title, &flags_copy);
```

Comment:
maybe no need to pass flags_copy here?
we could remove the UCS_CONFIG_PRINT_HEADER bit from *flags_p in ucs_config_parser_print_field instead of in ucs_config_parser_print_header.
also i'm not sure we need flags_copy in line 2194?